Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs

```diff
@@ -25,13 +28,32 @@ public void Process(ITelemetry item)
         _next.Process(item);
     }
 
-    private bool ShouldSend(ITelemetry item)
+    private static bool ShouldSend(ITelemetry item)
     {
         var dependency = item as DependencyTelemetry;
         if (dependency is not { Success: false }) return true;
 
-        dependency.Properties.TryGetValue("Error", out var error);
+        dependency.Properties.TryGetValue("Error", out var dependencyError);
         dependency.Properties.TryGetValue("Exception", out var exception);
-        return !_errors.Contains(error) && !_errors.Contains(exception);
+        dependency.Properties.TryGetValue("OrchardCore.ShellName", out var shellName);
+        var hasError = Errors.Exists(error => dependencyError?.StartsWith(error, StringComparison.OrdinalIgnoreCase) == true) ||
+            Errors.Exists(error => exception?.StartsWith(error, StringComparison.OrdinalIgnoreCase) == true) ||
+            exception?.Contains(GetTenantSqlErrorMessage(shellName)) == true;
+        var shouldSend = !hasError;
+
+        if (!shouldSend)
+        {
+            return false;
+        }
+
+        dependency.Properties.TryGetValue("OrchardCore.DataProtectionContainerName", out var dataProtectionContainerName);
+        dependency.Properties.TryGetValue("OrchardCore.MediaBlobStorageContainerName", out var mediaBlobStorageContainerName);
```

Comment:
I don't understand why we need to pass these from `AzureCheckExistsInitializer`. Why can't we check everything that's checked there here instead?

Reply:
You are right we don't need AzureCheckExistsInitializer and these container name checks now because we go with the generic check, it doesn't matter if we check the container name or not.